doc/ext/traffic-server.py

```diff
@@ -371,6 +374,37 @@ def xref_cleanup(app, env, node, contnode):
                 return node
     return;
 
+
+# get the branch this documentation is building for in X.X.x form
+with open('../configure.ac', 'r') as f:
+    contents = f.read()
+    match = re.compile('m4_define\(\[TS_VERSION_S],\[(.*?)]\)').search(contents)
+    autoconf_version = '.'.join(match.group(1).split('.', 2)[:2] + ['x'])
+
+# get the current branch the local repository is on
+git_branch = subprocess.check_output(['git', 'rev-parse', '--abbrev-ref', 'HEAD'])
+
+def make_github_link(name, rawtext, text, lineno, inliner, options={}, content=[]):
+    """
+    This docutils role lets us link to source code via the handy :ts:git: markup.
+    Link references are rooted at the top level source directory. All links resolve
+    to GitHub.
+
+    Examples:
+
+        To link to proxy/Main.cc:
+
+            Hi, here is a link to the proxy entry point: :ts:git:`proxy/Main.cc`.
+
+        To link to CONTRIBUTING.md:
+
+            If you want to contribute, take a look at :ts:git:`CONTRIBUTING.md`.
+    """
+    url = 'https://github.com/apache/trafficserver/blob/{}/{}'
+    ref = autoconf_version if autoconf_version == git_branch else 'master'
```

Comment:
Aha, that makes sense. I thought that we can use `git_branch` as is (without finding the parent), but I realized that it doesn't work because feature branches basically don't exist on GitHub. I agree finding the parent would be too complicated.